### PR TITLE
Feature/add property constraints decimal rounding for imperial and metric rounding

### PR DESCRIPTION
**Pull request by HOMAG-Robin-Katzenberger**

### CODE REVIEW

File: Applications/IntelliDivide/Contracts/Common/OptimizationBasePart.cs [hunk 1]
```diff
@@ -67,19 +67,19 @@ public OptimizationBasePart(UnitSystem unitSystem) : this()
         #region (2) Cutting / Nesting
 
         /// <inheritdoc />
-        [ValueDependsOnUnitSystem(BaseUnit.Millimeter)]
+        [ValueDependsOnUnitSystem(BaseUnit.Millimeter, PropertyDecimals.EdgebandingThicknessMillimeterDecimals, PropertyDecimals.InchDecimals)]
```
Comment (HOMAG-Boris-Wehrle) [inline Applications/IntelliDivide/Contracts/Common/OptimizationBasePart.cs:70]: Does this need to get specified in all cases? For Millimeter ist would be in most cases equal.

Comment (DaniSchaber) [reply Applications/IntelliDivide/Contracts/Common/OptimizationBasePart.cs]: As Robin is on holiday I updated it. Please check again

File: Applications/IntelliDivide/Contracts/Statistics/EdebandUsage.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ public class EdgebandUsage : IExtensibleDataObject, IContainsUnitSystemDependent
         /// Total length used for edgebands of this type (meters or feet)
         /// </summary>
         [JsonProperty(Order = 6)]
-        [ValueDependsOnUnitSystem(BaseUnit.Meter)]
+        [ValueDependsOnUnitSystem(BaseUnit.Meter, DecimalPrecision.TwoDecimalPlaces, DecimalPrecision.TwoDecimalPlaces)]
```
Comment (HOMAG-Boris-Wehrle) [inline Applications/IntelliDivide/Contracts/Statistics/EdebandUsage.cs:45]: I guess for "Meter" it is always 2 decimal places.

File: Applications/MaterialAssist/Contracts/Edgebands/EdgebandEntity.cs [hunk 3]
```diff
@@ -24,7 +24,7 @@ public class EdgebandEntity
         /// Gets or sets the thickness of the edgeband. The unit depends on the settings of the subscription (metric: mm, imperial:
         /// inch).
         /// </summary>
-        [ValueDependsOnUnitSystem(BaseUnit.Millimeter)]
+        [ValueDependsOnUnitSystem(BaseUnit.Millimeter, DecimalPrecision.TwoDecimalPlaces)]
```
Comment (HOMAG-Boris-Wehrle) [inline Applications/MaterialAssist/Contracts/Edgebands/EdgebandEntity.cs:27]: What do we use for inch?

File: Base/HomagConnect.Base.Contracts/Attributes/ValueDependsOnUnitSystemAttribute.cs [hunk 4]
```diff
@@ -5,10 +5,27 @@ namespace HomagConnect.Base.Contracts.Attributes;
...
+    /// Please pass a <see cref="metricDecimalPrecision" /> or <see cref="imperialDecimalPrecision" /> value if you want to
+    /// override the default decimals.
+    /// </summary>
+    /// <param name="baseUnit">The unit system of that property.</param>
+    /// <param name="metricDecimalPrecision">Defines how many decimals after comma are shown in metric system. Default is 1.</param>
+    /// <param name="imperialDecimalPrecision">
+    /// Defines how many decimals after comma are shown in imperial system. Default is
+    /// 3.
+    /// </param>
+    public ValueDependsOnUnitSystemAttribute(BaseUnit baseUnit, int metricDecimalPrecision = DecimalPrecision.OneDecimalPlace, int imperialDecimalPrecision = DecimalPrecision.ThreeDecimalPlaces)
```
Comment (HOMAG-Boris-Wehrle) [inline Base/HomagConnect.Base.Contracts/Attributes/ValueDependsOnUnitSystemAttribute.cs:19]: A overload which allows to switch off the rounding would in same cases be helpful.

bool roundValues = true

File: Applications/IntelliDivide/Contracts/Statistics/EdebandUsage.cs (hunk 2 above)
Comment (DaniSchaber) [reply Applications/IntelliDivide/Contracts/Statistics/EdebandUsage.cs:45]: Yes, but I set 1 (metric) and 3 (imperial) as defaults as they are used way more often.

File: Applications/MaterialAssist/Contracts/Edgebands/EdgebandEntity.cs (hunk 3 above)
Comment (DaniSchaber) [reply Applications/MaterialAssist/Contracts/Edgebands/EdgebandEntity.cs:27]: The default 3

File: Base/HomagConnect.Base.Contracts/Attributes/ValueDependsOnUnitSystemAttribute.cs (hunk 4 above)
Comment (DaniSchaber) [reply Base/HomagConnect.Base.Contracts/Attributes/ValueDependsOnUnitSystemAttribute.cs:19]: Sure, I can add that.

**Review by DaniSchaber — APPROVED** [review]

**Pull request merged**